pkg/storage/provider_no_send_test.go

```diff
@@ -3,162 +3,378 @@ package storage_test
 
-	// Step 3. Process the action using only the arguments, including the two previous txIDs, with IsNewTx set to false.
-	thirdProcessActionResult := fixture.ProcessAction(testusers.Alice.AuthID(), wdk.ProcessActionArgs{
-		IsNewTx:    false,
-		IsNoSend:   false,
-		SendWith:   []primitives.HexString{primitives.HexString(result.FirstTxID), primitives.HexString(result.SecondTxID)},
-		IsSendWith: true,
+		// when:
+		var noSendChangeOutpoints []wdk.OutPoint
+		for i := range noSendChainCount {
+			t.Logf("Creating NoSend tx #%d", i+1)
+			noSendChangeOutpoints = givenNoSend.CreateAndProcessNoSendAction(noSendChangeOutpoints)
+		}
+
+		// and:
+		// Call processAction using sendWith and IsNewTx set to false, including the two previous transactions in SendWithSlice.
+		sendWithProcessAction := givenNoSend.ProcessAction(wdk.ProcessActionArgs{
+			IsNewTx:    false,
+			IsNoSend:   false,
+			SendWith:   givenNoSend.NoSendTxsHexStrings(),
+			IsSendWith: true,
+		})
+
+		// then:
+		testabilities.NotDelayedResultsAsserter(sendWithProcessAction.NotDelayedResults).
+			ContainsTxsWithStatus(t, wdk.ReviewActionResultStatusSuccess, givenNoSend.NoSendTxs()...)
+
+		testabilities.SendWithResultsAsserter(sendWithProcessAction.SendWithResults).
+			ContainsTxsWithStatus(t, wdk.SendWithResultStatusUnproven, givenNoSend.NoSendTxs()...)
+
+		testabilities.
+			ThenDBState(t, activeStorage).
+			HasUserTransactionsByTxIDsWithStatus(testusers.Alice, wdk.TxStatusUnproven, givenNoSend.NoSendTxs()...)
+
+		testabilities.ThenFunds(t, testusers.Alice, activeStorage).
+			ShouldBeAbleToReserveSatoshis(
+				inputSatoshis +
+					-4 + // two no-send txs (2 sats each)
+					-7, // fee to create a new transaction with many inputs
+			)
 	})
 
-	testabilities.NotDelayedResultsAsserter(thirdProcessActionResult.NotDelayedResults).ContainsTxsWithStatus(t, wdk.ReviewActionResultStatusSuccess,
-		result.FirstTxID,
-		result.SecondTxID,
-	)
+	t.Run("three no-send txs, single initial UTXO, all noSendChange is used making chain of txs, no newTx when sendWith", func(t *testing.T) {
+		// given:
+		const inputSatoshis = 6
+		const noSendChainCount = 3
 
-	testabilities.SendWithResultsAsserter(thirdProcessActionResult.SendWithResults).ContainsTxsWithStatus(t, wdk.SendWithResultStatusUnproven,
-		result.FirstTxID,
-		result.SecondTxID,
-	)
+		given, cleanup := testabilities.Given(t)
+		defer cleanup()
 
-	testabilities.
-		ThenDBState(t, fixture.ActiveStorage()).
-		HasUserTransactionsByTxIDsWithStatus(testusers.Alice, wdk.TxStatusUnproven, result.FirstTxID, result.SecondTxID)
-}
+		activeStorage := given.Provider().GORM()
+		givenNoSend := testabilities.GivenNoSend(t, given, activeStorage, testusers.Alice)
 
-func TestNoSendSendWithScenario_SendWithNewTx(t *testing.T) {
-	// given:
-	const inputSatoshis = 99904
+		// and:
+		given.Faucet(activeStorage, testusers.Alice).TopUp(inputSatoshis) // Intentionally, create only one UTXO
```

Comment:
maybe new function to be explicit about intention?
```suggestion
		givenNoSend.UserHasSingleUTXOToSpend(inputSatoshis)
```

or:
```suggestion
		givenNoSend.UserReceivedSingleUTXO(inputSatoshis)
```